### ISSUE TITLE
Concurrency issue while compiling compiler bridge on a fresh machine?

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#5785]
## steps

Recently I ran into this while sending https://github.com/sbt/sbt/pull/5767/files.
If you change the `sbt/src/sbt-test/plugins/dotty-sandwich/test` to

```
> compile

> compile
```

and wipe out Travis CI cache you might be able to reproduce what I saw.

## problem

The problem you come across may be somewhat arbitrary but one example is "Cannot redefine component"

```
[info] Running plugins/dotty-sandwich
[info] [info] welcome to sbt 1.4.0-SNAPSHOT (AdoptOpenJDK Java 11.0.1)
[info] [info] loading settings for project global-plugins from metals.sbt ...
[info] [info] loading global plugins from /tmp/sbt_d7daa57/global/plugins
[info] [info] loading settings for project sbt_d7daa57-build from plugins.sbt ...
[info] [info] loading project definition from /tmp/sbt_d7daa57/project
[info] [info] loading settings for project root from build.sbt ...
[info] [info] set current project to root (in build file:/tmp/sbt_d7daa57/)
[info] [info] compiling 1 Scala source to /tmp/sbt_d7daa57/foo-core/target/scala-2.13/classes ...
[info] [info] compiling 1 Scala source to /tmp/sbt_d7daa57/bar-core/target/scala-0.23/classes ...
[info] [info] Non-compiled module 'compiler-bridge_2.13' for Scala 2.13.4-bin-aeee8f0. Compiling...
[info] [info] done compiling
[info] [info] compiling 1 Scala source to /tmp/sbt_d7daa57/bar-app/target/scala-2.13/classes ...
[info] [info] Non-compiled module 'compiler-bridge_2.13' for Scala 2.13.4-bin-aeee8f0. Compiling...
[info] [info]   Compilation completed in 18.539s.
[info] [info]   Compilation completed in 24.223s.
[info] [info] Attempting to fetch org.scala-sbt:compiler-bridge_2.13:1.4.0-M10.
[info] [error] ## Exception when compiling 1 sources to /tmp/sbt_d7daa57/foo-core/target/scala-2.13/classes
[info] [error] sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$1$1: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: Cannot redefine component. ID: sbt.internal.scriptedtest.ScriptedLauncher$1@73f23e8e, files: /tmp/sbt_159a0475/org.scala-sbt-compiler-bridge_2.13-1.4.0-M10-bin_2.13.4-bin-aeee8f0__55.0.jar
[info] [error] 
[info] [error]            
[info] [info] done compiling
[info] [error] sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$1$1: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: Cannot redefine component. ID: sbt.internal.scriptedtest.ScriptedLauncher$1@73f23e8e, files: /tmp/sbt_159a0475/org.scala-sbt-compiler-bridge_2.13-1.4.0-M10-bin_2.13.4-bin-aeee8f0__55.0.jar
[info] [error] Caused by: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: Cannot redefine component. ID: sbt.internal.scriptedtest.ScriptedLauncher$1@73f23e8e, files: /tmp/sbt_159a0475/org.scala-sbt-compiler-bridge_2.13-1.4.0-M10-bin_2.13.4-bin-aeee8f0__55.0.jar
[info] [error] (fooCore / Compile / compileIncremental) sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$1$1: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: sbt.internal.scriptedtest.ScriptedLauncher$3$1$1$1$foo: Cannot redefine component. ID: sbt.internal.scriptedtest.ScriptedLauncher$1@73f23e8e, files: /tmp/sbt_159a0475/org.scala-sbt-compiler-bridge_2.13-1.4.0-M10-bin_2.13.4-bin-aeee8f0__55.0.jar
[error] x plugins/dotty-sandwich 
[error]  Cause of test exception: {line 1}  Command failed: compile failed
```

## expectation

Some form of concurrency control and logging around this.

**Comment by eugeniyk** [comment]
Keep having this daily:
![image](https://user-images.githubusercontent.com/3932426/111447991-93aa4b80-8740-11eb-85b8-1fe90d78fa69.png)

sbt 1.4.6, scala 2.12.x

**Comment by rfan-debug** [comment]
I am also having the same issue:

sbt 1.3.13, scala 2.12.x, open-jdk 11. 

I am kinda not sure if the issue came from the jdk 11 upgrade. We used to use JDK8 and there was no such an issue.

**Comment by SethTisue** [comment]
but has it ever happened with the current release (sbt 1.5.4)? a fix has been merged, but it's expected that you don't have the fix if you are continuing to run old versions like 1.3.13 and 1.4.6

**Comment by rfan-debug** [comment]
My datapoint is: after switching back to openjdk8, this issue does not happen any more. 

My setting: sbt 1.3.13, scala 2.12.13, open-jdk8

**Comment by eed3si9n** [comment]
Yea. According to https://github.com/sbt/sbt/commit/3c955abea626141b5e959a2075f2d58e116f93df, the earliest fixed version is 1.5.0, so please try the current stable first.
